extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/FloatPushdownHandler.java

```diff
@@ -0,0 +1,140 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.parquet.table.location;
+
+import io.deephaven.engine.table.impl.select.FloatRangeFilter;
+import io.deephaven.engine.table.impl.select.MatchFilter;
+import io.deephaven.util.QueryConstants;
+import io.deephaven.util.type.ArrayTypeUtils;
+import org.apache.commons.lang3.mutable.MutableObject;
+import org.apache.parquet.column.statistics.Statistics;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+
+final class FloatPushdownHandler {
+
+    /**
+     * Verifies that the statistics range intersects the range defined by the filter.
+     */
+    static boolean maybeOverlaps(
+            @NotNull final FloatRangeFilter floatRangeFilter,
+            @NotNull final Statistics<?> statistics) {
+        // Skip pushdown-based filtering for nulls and NaNs to err on the safer side instead of adding more complex
+        // handling logic.
+        // TODO (DH-19666): Improve handling of nulls
+        final float dhLower = floatRangeFilter.getLower();
+        final float dhUpper = floatRangeFilter.getUpper();
+        if (Float.isNaN(dhLower) || Float.isNaN(dhUpper) ||
+                dhLower == QueryConstants.NULL_FLOAT || dhUpper == QueryConstants.NULL_FLOAT) {
+            return true;
+        }
```

Comment:
Can we actually put a NaN into a Range filter? What does it match if we do?

Reply:
NaN is included in the upper range; see `DoubleRangeFilter#gt` / `DoubleRangeFilter#geq` have upperInclusive of NaN. There is some reasoning we might want to change this in the future, but for now pushdown has to be consistent w/ engine expectations.